server/Models/getRestaurantInfo.js

```diff
@@ -0,0 +1,68 @@
+const pg = require('../db');
+
+const getRestaurantInfo = (req, res) => {
+  const { restaurantId } = req.params;
+
+  const { sort, keyword, star } = req.body;
+  let orderByStr = '';
+  let selectStr = orderByStr;
+  let sortDirection = '';
+  let keywordStr = '';
+  let starStr = '';
+
+  if (sort === 'highest rating') {
+    orderByStr = 'CAST((reviews.foodrating+reviews.servicerating+reviews.ambiencerating+reviews.valuerating) AS FLOAT)/4';
+    sortDirection = ' DESC,';
+  }
+
+  if (sort === 'lowest rating') {
+    orderByStr = 'CAST((reviews.foodrating+reviews.servicerating+reviews.ambiencerating+reviews.valuerating) AS FLOAT)/4';
+    sortDirection = ' ASC,';
+  }
+
+  if (keyword !== undefined) {
+    for (let i = 0; i < keyword.length; i ++) {
+      keywordStr += `AND reviews.reviewText LIKE '%${keyword[i]}%' `;
+    }
+
+    keywordStr = keywordStr.slice(0, -1);
+  }
+
+  if (orderByStr !== '') {
+    selectStr = ', ' + orderByStr + ' AS avg_overal_from_reviews';
+  }
+
+  if (star !== undefined) {
+    starStr = `AND CAST((reviews.foodrating+reviews.servicerating+reviews.ambiencerating+reviews.valuerating) AS FLOAT)/4 >= ${star} AND CAST((reviews.foodrating+reviews.servicerating+reviews.ambiencerating+reviews.valuerating) AS FLOAT)/4 < ${star + 1}`;
```

Comment:
basically here you will replace the whole `CAST((reviews.foodrating+reviews.servicerating+reviews.ambiencerating+reviews.valuerating) AS FLOAT)/4` 
with `avg_overal_from_reviews`

Reply:
Since 'WHERE' is performed prior to the alias declaration in 'SELECT', the alias is not recognized in the 'WHERE' statement. But I was able to switch the 'cast' statement with the alias in the 'ORDER BY' statement.